src/transformers/tokenization_utils_base.py

```diff
@@ -1728,9 +1728,16 @@ def apply_chat_template(
             output is ready to pass to the model, either directly or via methods like `generate()`.
         """
 
-        if hasattr(conversation, "messages"):
-            # Indicates it's a Conversation object
-            conversation = conversation.messages
+        if return_dict is None and tokenize:
+            logger.warning_once(
+                "In version 4.40, `return_dict` will be set to `True` by default. "
+                "Please explicitly set `return_dict` to `False` to maintain the current behaviour, "
+                "or set it to `True` to get the new behaviour immediately."
+            )
+            return_dict = False
+
+        if return_dict and not tokenize:
+            raise ValueError("return_dict=True is incompatible with tokenize=False")
```

Comment:
```suggestion
            raise ValueError("`return_dict=True` is incompatible with `tokenize=False`")
```
also explaining why should help users